internal/attestation/crafter/crafter.go

```diff
@@ -427,17 +422,6 @@ func (c *Crafter) ResolveEnvVars(strict bool) error {
 	return nil
 }
 
-func notResolvedVars(resolved map[string]string, wantList []string) []string {
```

Comment:
Ok, so I think I remember.

With the new code, the error is being returned by the method that resolves the variables, not by this auxiliary method.

I see that the new errors only show 1 missing environment variable, even in the case multiple vars are missing. Maybe we could leave this method but then its invocation happens in the ResolveEnvRun method? Also, I think the specific now changes slightly, because if it's an error, it means it's only considering the variables that are not resolved and that were required. Do you also see this point the same way as me?

Reply:
Agreed, let's not use `notResolvedVar` as per your snippet.

To be able to know what are all the non-resolved vars as part of `ResolveEnvVars()`, we could return an array of errors. Similar to https://github.com/kubernetes/apimachinery/blob/fa98d6eaedb4caccd69fc07d90bbb6a1e551f00f/pkg/api/validation/objectmeta.go#L69

That one returns `ErrorList` which is `[]*Error`. Then, instead of checking `err == nil` we will check for its length.

What do you think?

Reply:
I agree with you, meaning that the error should be returned by ResolveEnvVars().

I'll work on an implementation.
Thanks for the feedback.